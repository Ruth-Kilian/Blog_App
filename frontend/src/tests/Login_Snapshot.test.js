/* Snapshot test for Login Form*/

// import dependencies
import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom"; // Import BrowserRouter
import LoginForm from "../components/Login/Login";

// define the test
test("renders LoginForm correctly", () => {
  // render the component
  const { container } = render(
    // wrap component with Router
    <Router>
      <LoginForm />
    </Router>
  );
  // assert the test
  expect(container).toMatchSnapshot();
});
