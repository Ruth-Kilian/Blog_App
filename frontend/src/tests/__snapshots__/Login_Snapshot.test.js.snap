// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginForm correctly 1`] = `
<div>
  <div
    class="d-flex justify-content-center align-items-center container"
    style="height: 100vh;"
  >
    <div
      class="login-box shadow p-5"
    >
      <h2>
        Login
      </h2>
      <p
        style="margin-bottom: 20px; font-size: 15px;"
      >
        Don't have an account?
         
        <a
          href="/signup"
          style="margin-bottom: 20px; font-size: 11px; text-decoration: underline; font-weight: bold;"
        >
          Sign Up
        </a>
      </p>
      <form
        class=""
      >
        <div>
          <label
            class="form-label"
            for="username"
          >
            Username
          </label>
          <input
            class="form-control"
            id="username"
            placeholder="Enter username"
            style="margin-bottom: 15px;"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            class="form-label"
            for="password"
          >
            Password
          </label>
          <input
            class="form-control"
            id="password"
            placeholder="Enter password"
            type="password"
            value=""
          />
        </div>
        <button
          class="btn btn-primary"
          style="margin-top: 15px;"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
